sdcm/nemesis.py

```diff
@@ -445,18 +445,79 @@ def disrupt_nodetool_cleanup(self):
+        self._set_current_disruption('ModifyTableProperties%s %s' % (name.capitalize(), self.target_node))
+        cmd = "ALTER TABLE {keyspace}.{table} WITH {name} = {val};".format(**locals())
+        self._run_in_cqlsh(cmd)
+
     def disrupt_modify_table_comment(self):
-        self._set_current_disruption('ModifyTableProperties %s' % self.target_node)
-        comment = ''.join(random.choice(string.ascii_letters) for i in xrange(24))
-        cmd = "ALTER TABLE keyspace1.standard1 with comment = '{}';".format(comment)
-        self.target_node.remoter.run('cqlsh -e "{}" {}'.format(cmd, self.target_node.private_ip_address), verbose=True)
+        prop_val = ''.join(random.choice(string.ascii_letters) for _ in xrange(24))
+        self._modify_table_property(name="comment", val="'%s'" % prop_val)
 
     def disrupt_modify_table_gc_grace_time(self):
-        self._set_current_disruption('ModifyTableProperties %s' % self.target_node)
-        gc_grace_seconds = random.choice(xrange(216000, 864000))
-        cmd = "ALTER TABLE keyspace1.standard1 with comment = 'gc_grace_seconds changed' AND" \
-              " gc_grace_seconds = {};".format(gc_grace_seconds)
-        self.target_node.remoter.run('cqlsh -e "{}" {}'.format(cmd, self.target_node.private_ip_address), verbose=True)
+        self._modify_table_property(name="gc_grace_seconds", val=random.randint(216000, 864000))
+
+    def disrupt_modify_table_caching(self):
+        prop_val = dict(
+            keys=random.choice(["NONE", "ALL"]),
+            rows_per_partition=random.choice(["NONE", "ALL", random.randint(1, 10000)])
+        )
+        self._modify_table_property(name="caching", val=str(prop_val))
+
+    def disrupt_modify_table_bloom_filter_fp_chance(self):
+        self._modify_table_property(name="bloom_filter_fp_chance", val=random.random()/2)
+
+    def disrupt_modify_table_compaction(self):
+        # TODO: Sub-properties for each of compaction strategies should also be tested
+        strategies = ("SizeTieredCompactionStrategy", "DateTieredCompactionStrategy",
+                      "TimeWindowCompactionStrategy", "LeveledCompactionStrategy")
+        prop_val = {"class": random.choice(strategies)}
+        self._modify_table_property(name="compaction", val=str(prop_val))
+
+    def disrupt_modify_table_compression(self):
+        algos = ("",  # no compression
+                 "LZ4Compressor",
+                 "SnappyCompressor",
+                 "DeflateCompressor")
+        algo = random.choice(algos)
+        prop_val = {"sstable_compression": algo}
+        if algo:
+            prop_val["chunk_length_kb"] = random.choice(["4K", "64KB", "1M"])
+            prop_val["crc_check_chance"] = random.random()
+        self._modify_table_property(name="compression", val=str(prop_val))
+
+    def disrupt_modify_table_crc_check_chance(self):
+        self._modify_table_property(name="crc_check_chance", val=random.random())
```

Comment:
Doesn't the value have range?

Reply:
it is a probability :) it can be any float in semi-open range [0.0, 1.0)